modules/docs/docgen/docParser.ts

```diff
@@ -1227,11 +1253,9 @@ export function getDefaultsFromObjectBindingParameter(
   if (t.isObjectBindingPattern(node.name)) {
     return node.name.elements.reduce((result, element) => {
       if (t.isBindingElement(element) && t.isIdentifier(element.name) && element.initializer) {
-        // this might change in the future. If the default value isn't a literal expression, it is
-        // much harder to deal with. For example, `{a = 'a'}` instead of `{a = getA(someInput)}`.
-        // Until we figure out how to mark defaults for non-literals, this might be a limitation.
-        if (ts.isLiteralExpression(element.initializer)) {
-          result[element.name.escapedText as string] = parser.getValueFromNode(element.initializer);
+        const defaultValue = getValidDefaultFromNode(parser, element.initializer);
+        if (defaultValue) {
+          result[element.name.text] = parser.getValueFromNode(element.initializer);
```

Comment:
This stood out to me that we're checking for a `defaultValue` but not assigning it here, like we do in the recent change on line 238 in `componentParser.ts`.

Reply:
Good catch